shared/models/git_repo.rb

```diff
@@ -294,6 +294,7 @@ def commit_changes!(commit_message: nil, file_to_commit: nil, repo_auth: self.re
           logger.debug("done commit_changes! #{self.git_config.full_name}")
         end
       end
+      wait_for_task_to_complete(task: task) unless async
```

Comment:
Instead of this, could we make use of block based Ruby code? Providing a callback?

Reply:
The problem is that this is being triggered by a web request, there're chances that the commit does not finish after the project being created and having `user minuscorp@gmail.com doesn't have access to a project with id`.
```ruby
authenticated_controller_base.rb in user_project_with_id
raise "user #{self.user.email} doesn't have access to a project with id `#{project_id}`" if project.nil?
```